Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestResultsValidationService.cs

```diff
@@ -1,87 +1,44 @@
-        IContestsActivityService activityService,
-        IParticipantsDataService participantsData,
-        IDatesService datesService,
-        IUserProviderService userProvider)
-    {
-        this.notDefaultValueValidationHelper = notDefaultValueValidationHelper;
-        this.lecturersInContestsBusiness = lecturersInContestsBusiness;
-        this.activityService = activityService;
-        this.participantsData = participantsData;
-        this.datesService = datesService;
-        this.userProvider = userProvider;
-    }
-
-    public async Task<ValidationResult> GetValidationResult((ContestDetailsServiceModel?, bool, bool) item)
+    public ValidationResult GetValidationResult((IContestActivityServiceModel?, bool, bool, bool) item)
     {
-        var (contest, fullResults, isOfficial) = item;
+        var (contest, fullResults, isOfficial, isUserAdminOrLecturerInContest) = item;
 
-        var contestNullValidationResult = this.notDefaultValueValidationHelper.ValidateValueIsNotDefault(contest, nameof(contest));
+        var contestNullValidationResult = notDefaultValueValidationHelper.ValidateValueIsNotDefault(contest, nameof(contest));
 
         if (!contestNullValidationResult.IsValid)
         {
             return contestNullValidationResult;
         }
 
-        var user = this.userProvider.GetCurrentUser();
-
-        var isUserAdminOrLecturer = user != null && (user.IsAdmin || await this.lecturersInContestsBusiness
-            .IsCurrentUserAdminOrLecturerInContest(contest?.Id));
-
-        if (fullResults && !user!.IsAdminOrLecturer)
+        if (isUserAdminOrLecturerInContest)
         {
-            return ValidationResult.Invalid(ValidationMessages.Participant.NoPrivilegesForContestResults);
+            return ValidationResult.Valid();
         }
 
-        if (isUserAdminOrLecturer)
+        if (fullResults && !isUserAdminOrLecturerInContest)
         {
-            return ValidationResult.Valid();
+            return ValidationResult.Invalid(ValidationMessages.Participant.NoPrivilegesForContestResults);
         }
 
-        var contestIsVisible = contest?.IsVisible == true || contest?.VisibleFrom <= this.datesService.GetUtcNow();
-
-        if (!contestIsVisible || contest!.IsDeleted)
+        if (!contest!.IsVisible)
         {
             return ValidationResult.Invalid(ValidationMessages.Participant.ResultsNotVisibleForContest);
         }
 
-        var contestActivity = await this.activityService.GetContestActivity(contest!.Map<ContestForActivityServiceModel>());
-
-        var contestHasStartDateAndEndTimeIsBeforeNow =
-            contest.StartTime.HasValue &&
-            contest.EndTime.HasValue &&
-            contest.EndTime <= this.datesService.GetUtcNow();
-
-        if (isOfficial && !contestHasStartDateAndEndTimeIsBeforeNow && await this.participantsData
-                .GetByContestByUserAndByIsOfficial(contest.Id, user!.Id, isOfficial) == null)
```

Comment:
Why did you remove this validation?

Reply:
It's inlcuded in the contestActivity result.